drivers/sensor/lis2ds12/lis2ds12.c

```diff
@@ -267,32 +240,121 @@ static int lis2ds12_init(const struct device *dev)
 
 	k_busy_wait(100);
 
-	LOG_DBG("chip id 0x%x", chip_id);
+	LOG_DBG("%s: chip id 0x%x", dev->name, chip_id);
 
 #ifdef CONFIG_LIS2DS12_TRIGGER
-	if (lis2ds12_trigger_init(dev) < 0) {
-		LOG_ERR("Failed to initialize triggers.");
-		return -EIO;
+	ret = lis2ds12_trigger_init(dev);
+	if (ret < 0) {
+		LOG_ERR("%s: Failed to initialize triggers", dev->name);
+		return ret;
 	}
 #endif
 
 	/* set sensor default odr */
 	ret = lis2ds12_set_odr(dev, 12);
 	if (ret < 0) {
-		LOG_ERR("odr init error (12.5 Hz)");
+		LOG_ERR("%s: odr init error (12.5 Hz)", dev->name);
 		return ret;
 	}
 
 	/* set sensor default scale */
 	ret = lis2ds12_set_range(dev, CONFIG_LIS2DS12_FS);
 	if (ret < 0) {
-		LOG_ERR("range init error %d", CONFIG_LIS2DS12_FS);
+		LOG_ERR("%s: range init error %d", dev->name, CONFIG_LIS2DS12_FS);
 		return ret;
 	}
 
 	return 0;
 }
 
-DEVICE_DT_INST_DEFINE(0, lis2ds12_init, NULL,
-		    &lis2ds12_data, &lis2ds12_config, POST_KERNEL,
-		    CONFIG_SENSOR_INIT_PRIORITY, &lis2ds12_api_funcs);
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "LIS2DS12 driver enabled without any devices"
+#endif
+
+/*
+ * Device creation macro, shared by LIS2DS12_DEFINE_SPI() and
+ * LIS2DS12_DEFINE_I2C().
+ */
+
+#define LIS2DS12_DEVICE_INIT(inst)					\
+	DEVICE_DT_INST_DEFINE(inst,					\
+			    lis2ds12_init,				\
+			    NULL,					\
+			    &lis2ds12_data_##inst,			\
+			    &lis2ds12_config_##inst,			\
+			    POST_KERNEL,				\
+			    CONFIG_SENSOR_INIT_PRIORITY,		\
+			    &lis2ds12_driver_api);
+
+/*
+ * Instantiation macros used when a device is on a SPI bus.
+ */
+
+#ifdef CONFIG_LIS2DS12_TRIGGER
+#define LIS2DS12_CFG_IRQ(inst) \
+	.gpio_int = GPIO_DT_SPEC_INST_GET(inst, irq_gpios),
+#else
+#define LIS2DS12_CFG_IRQ(inst)
+#endif /* CONFIG_LIS2DS12_TRIGGER */
+
+#define LIS2DS12_SPI_OPERATION (SPI_WORD_SET(8) |			\
+				SPI_OP_MODE_MASTER |			\
+				SPI_MODE_CPOL |				\
+				SPI_MODE_CPHA)				\
+
+#define LIS2DS12_CONFIG_SPI(inst)					\
+	{								\
+		.ctx = {						\
+			.read_reg =					\
+			   (stmdev_read_ptr) stmemsc_spi_read,		\
+			.write_reg =					\
+			   (stmdev_write_ptr) stmemsc_spi_write,	\
+			.handle =					\
+			   (void *)&lis2ds12_config_##inst.stmemsc_cfg,	\
+		},							\
+		.stmemsc_cfg.spi = {					\
+			.bus = DEVICE_DT_GET(DT_INST_BUS(inst)),	\
+			.spi_cfg = SPI_CONFIG_DT_INST(inst,		\
```

Comment:
`SPI_DT_SPEC_INST_GET` should be used

Reply:
It seems fine, as it is a code semplification. But I have to change it into stmemsc_cfg_spi structure, which is common to many ST sensor now. So, I have to do it as a preliminary commit for those sensors first, and I will do it in this PR itself. Sounds good?

Reply:
It’s ok, just raised as it’s always better to use existing facilities when available.

Reply:
> It’s ok, just raised as it’s always better to use existing facilities when available.

Yes, correct. THanks for pointing this out. The stmemsc common APIs are now quite simpler.